src/AIAPI/MainGame/GameAPI.Hooks.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections;
+// using ActionGame;
+using HarmonyLib;
+using AIProject.SaveData;
+using Manager;
+using AIProject;
+using Sirenix.Serialization;
+
+namespace KKAPI.MainGame
+{
+    public static partial class GameAPI
+    {
+        private class Hooks
+        {
+            public static int lastCurrentDay = 1;//Always starts at 1
+
+            public static void SetupHooks()
+            {
+                Harmony.CreateAndPatchAll(typeof(Hooks));
+            }
+
+            //TODO SaveData.Load only has 1 param in AI and it's the path, does not include filename like KK
+            [HarmonyPostfix]
+            [HarmonyPatch(typeof(SaveData), nameof(SaveData.Load), new[] { typeof(string) })]
+            public static void LoadHook(string fileName)
+            {
+                OnGameBeingLoaded("", fileName);
+            }
+
+            [HarmonyPrefix]
+            [HarmonyPatch(typeof(SaveData), nameof(SaveData.SaveFile), new[] { typeof(string) })]
+            public static void SaveHook(string path)
+            {
+                OnGameBeingSaved(path, "");
+            }
+
+            [HarmonyPostfix]
+            [HarmonyPatch(typeof(HScene), "InitCoroutine")]
+            public static void HScene_InitCoroutine(HScene __instance)
+            {
+                OnHStart(__instance);
+            }      
+
+            [HarmonyPostfix]
+            [HarmonyPatch(typeof(HScene), "EndProc")]
+            public static void HScene_EndProc(HScene __instance)
+            {
+                OnHEnd(__instance);
+            }
+
+            [HarmonyPostfix]
+            [HarmonyPatch(typeof(EnvironmentSimulator), nameof(EnvironmentSimulator.SetTimeZone), typeof(AIProject.TimeZone))]
+            public static void CycleChangeTypeHook(AIProject.TimeZone zone)
+            {
+                OnPeriodChange(zone);
+            }
+
+            [HarmonyPostfix]
+            [HarmonyPatch(typeof(EnviroSky), "SetGameTime")]
+            public static void CycleChangeDayHook(EnviroSky __instance)
```

Comment:
KK Cycle is gone.  AI seems to be using EnviroSky to keep track of time and days.

Reply:
You don't have to follow the KK API so closely if it doesn't make sense in context of AI. Plugins will need to have major differences in their gameplay code anyways so it might be better to have different event names to make things clearer, or have a different set of events.